documentation/Data-View-Filters.md

```diff
@@ -0,0 +1,385 @@
+
+#### How is Filter data generated?
+
+There is a script, `generateFilterDataFields.ts` that can create and update all filter data columns. The script scans the system for existing filter data columns and updates them or creates new ones based on what's in the `data_view_column_definitions` table. It can also (optionally) regenerate ALL values for existing filter_data fields. (Note that if the column definitions have been removed, the generated "filterData" fields will be removed from the database.)
+
+Currently, the only time the app automatically runs it is as an option on the [modifyRecord](List-of-Action-plugins.md#modify-record) action, which ensures that new records automatically have filter data generated for them.
+
+However, there is an endpoint that can be called as required to regenerate all filter data throughout the system. You would normally want to run this after updating or adding filter data column definitions to ensure all data is up-to-date.
+
+The API is:
+
+```
+/admin/generate-filter-data-fields?<query>
+```
+
+Query parameters:
+
+- `table`: the name of the database table to update filter_data for. If not provided, it will update *all* filter data in the system
+- `fullUpdate` (`true`/`false`): if `true`, the script will re-compute (i.e. will re-evaluate every record) the filter_data for all existing filter data as well as generating values for missing data. If `false`, it will only update records where all filter data columns are `null` (i.e. these should be *new* records). By default, `fullUpdate` is `false` when a `table` is specified (this is what `modifyRecord` uses), but `true` when no table is specified (i.e. all tables).
+
+
+### Filter columns
+
+The columns that will appear as available filters for a given data view are defined in `filter_include_columns` and `filter_exclude_columns`. These work the same way as the table view and detail view [include/exclude lists](Data-View.md#data_view-table), but with the following differences:
+
+- if nothing is specified (i.e. `null` value), the defaults will be the same as the table view columns that are being returned. This is because, by default, we expect the available filters to be for the same data as the table columns.
+- Similarly, the special value `...` matches all *table* columns, not the full set of columns. So it becomes fairly straightfoward to specify a filter set that closely matches the table columns but with, for example, one extra column and one exclusion. It's a good idea to explicitly exclude columns for which the data can't be processed by one of the standard filters -- however, using mapped "filter data" columns (see [above](#handling-complex-data-structures)) it's possible to get meaningful, filter-able content from almost any type of data.
```

Comment:
"matches all *table* columns, not the full set of columns." Not entirely sure on what "full set of columns" means

Reply:
Sorry to be pick, but this text seems a bit long, I would suggest changing this phrase: "So it becomes fairly straightfoward to specify a filter set that closely matches the table columns but with, for example, one extra column and one exclusion."

 to something more explanatory or an example to how to use this special value

Reply:
I've already explained how `...` works in the Data View documentation -- this just clarifies how it's subtly different.

Reply:
Yeah, I think it's clear when taken in the context of what came before it. I'm open to explaining it better, but I just need to know specifically which part is not clear. (Tell me in person if you'd like further clarification).